otwcode/otwarchive
# AO3-4559: Languages can be sorted alphabetically by optional sortable_name

## Issue

https://otwarchive.atlassian.net/browse/AO3-4559

## Purpose

Allows admins to set an optional name to use for alphabetization. If no name is set, it should continue to fall back to the abbreviation (language.short).

## Testing

How can the Archive's QA team verify that this is working as you intended? (If you have access, please copy this into the JIRA ticket for them!)

1. Log in as admin
2. Go to languages page at http://test.ao3.org/languages
3. Choose "Edit" for a language, e.g. Suomi
4. Enter an optional name to sort on (e.g. Suomi) in a text field labeled "Name for alphabetical sorting"
5. Update Language
6. Voila ! Suomi is now with the other S-languages on the Language menu

The language menu is found in many places on AO3:
- http://test.ao3.org/admin_posts
- http://test.ao3.org/admin_posts/### --> individual admin post w/ translations listed on top of post
- http://test.ao3.org/works/search - work search page
- http://test.ao3.org/works/new - new work page (2 places: "Choose a language" dropdown in Associations section, and check "This work is a remix, a translation, a podfic, or was inspired by another work" option for a separate Language dropdown)
- http://test.archiveofourown.org/users/USERNAME/works/edit_multiple - edit multiple works page
- http://test.ao3.orgtags/TAGNAME/works - filters for works listing
- http://test.ao3.org/tags/TAGNAME/bookmarks - filters for bookmarks listing
- http://test.ao3.org/admin_posts/new - admin post form (needs admin access)
- https://test.ao3.org/languages - language list
- https://test.archiveofourown.org/abuse_reports/new - new abuse report
- https://test.archiveofourown.org/support - new support request

## Credit

*What name do you want us to use to credit your work in the [Archive of Our Own's Release Notes](http://archiveofourown.org/admin_posts?tag=1)?*
cyrilcee

*What pronouns do you prefer (she/her, he/him, zie/hir etc)?*
she/her
